### DISCUSSION TITLE
Combining edge/ corner finding and angle measurement

### DISCUSSION THREAD

**Discussion by SebastianM81** [terjeio/ioSender#305] [Ideas]
Hey Terjeio, 
first of all I wanna thank you for the incredible work that you do with ioSender. It really makes my life easier :)

To make the tool even more awesome, I might have an Idea for improvement...

I realized that its not possible with the current implementation to precisly touch-off a workpiece that is skewed / rotated on the baseboard.
Let me explain: 
Say I have an aluminum block that is rotated by 45degrees (extreme example) and I wanna to the edge / corner fiding. IOsender will probe one point per side, but this will always lead to a wrong (assumed) corner. (see example in first picture). 
Then, I would do the angle probing, but as the origin of the part is already wrong, I will end up with right shape and angle, but at the wrong position... (see screenshots 2 and 3).
And depending on the positions of the probing points, I can end up with several different results for the same workpiece (scenario 1 vs 2 in the pictures).

I have used Estlcam a few years ago, and Estlcam has another approach for this... when doing the Edgefinding, Estlcam probes 2 points on the first side and 1 point on the second side... so it basically combines the corner finding and angle measurement. 

I hope you understand what I mean. 
Again - I love Iosender, but if you want to do som really precise cuttings, this offset that I described is a bit annoying. 

Regards!

<img width="833" alt="1" src="https://github.com/terjeio/ioSender/assets/137636385/87964d76-36b7-4c3c-a835-74ca93aedbc7">

<img width="822" alt="2" src="https://github.com/terjeio/ioSender/assets/137636385/9e2f5a73-4fb2-49d0-aa12-708f05ba730a">

<img width="822" alt="3" src="https://github.com/terjeio/ioSender/assets/137636385/c4fce3d4-35ad-4c57-a17d-8208ac225e26">

**Comment by terjeio** [comment]
New [edge version](http://www.io-engineering.com/downloads) up that that has initial support for an option to probe a third point to determine and set X0,Y0:

![image](https://github.com/terjeio/ioSender/assets/20260062/4f5c4d96-17a7-4fa0-af02-060c703dc8ba)

The option is only available/usable for a right angled lower left corner, I do not know if there is any need to add it for the other corners and/or arbitrarily angled corners - I have left that out for now as it will add a lot more complexity.

Functionality should be similar as shown in [this Estlcam video](https://youtu.be/bhZdbgM6S70?t=67).

Be aware that I have _not_ tested this extensively, so please be careful whe using the option.
